openedx/core/djangoapps/catalog/utils.py

```diff
@@ -210,5 +245,111 @@ def get_run_marketing_url(course_key, user):
     Returns:
         string, the marketing URL, or None if no URL is available.
     """
-    course_run = get_course_run(course_key, user)
-    return course_run.get('marketing_url')
+    course_marketing_urls = get_run_marketing_urls([course_key], user)
+    return course_marketing_urls.get(unicode(course_key))
+
+
+def get_run_marketing_urls(course_keys, user):
+    """
+    Get course run marketing URLs from the course catalog service against course keys.
+
+    Arguments:
+        course_keys ([CourseKey]): A list of Course key object identifying the run whose data we want.
+        user (User): The user to authenticate as when making requests to the catalog service.
+
+    Returns:
+        dict of run marketing URLs against course keys
+    """
+    course_marketing_urls = {}
+    course_catalog_dict = get_course_runs(course_keys, user)
```

Comment:
this name is confusing to me, if this contains course run information coming from the catalog i would name it course_run_dict or catalog_course_run_dict (this name is similar to people just calling the catalog course catalog sometimes)